spec/src/main/asciidoc/WebSocket.adoc

```diff
@@ -593,10 +593,32 @@ match [WSC-3.1.2-1].
 [[extension-modification]]
 ==== Extension Modification
 
-In the opening handshake, the client supplies a list of extensions that
-it would like to use. The default server configuration selects from
-those extensions the ones it supports, and places them in the same order
-as requested by the client [WSC-3.1.3-1].
+The starting point to determine the extensions and parameters the default server
+configuration must return in the opening handshake response is the list of
+extensions and parameters providing by the client in the opening handshake. The
+order in which extensions are presented in the opening handshake must be
+maintained as the order determines the order in which the extensions will
+process WebSocket messages.
+
+Any extension whose name is not included in the list of extensions returned by
+`ServerContainer.getInstalledExtensions()` must be excluded from the opening
+handshake response. Implementation and configuration of supported extensions is
+container specific.
+
+The client may include the same extension multiple times. This signifies that
+different configurations of the extension are acceptable with the variations
+listed in preference order from most preferred to least preferred. The server
+should include its preferred configuration in the opening handshake response.
+If none of the requested configurations are acceptable, the extension is not
+included in the opening handshake response [WSC-3.1.3-1].
```

Comment:
The server may also add or drop extension parameters as permitted by that extension’s specification.